lib/graphql/input_object_type.rb

```diff
@@ -75,16 +75,14 @@ def coerce_non_null_input(value)
 
       arguments.each do |input_key, input_field_defn|
         field_value = value[input_key]
-        field_value = input_field_defn.type.coerce_input(field_value)
 
-        # Try getting the default value
-        if field_value.nil?
```

Comment:
Is there a discrepancy between arguments and input objects, though?
- If `coerce_input` returns nil for a field argument, the query fails
- If `coerce_input` returns nil for an input object, the application code receives the default value